p-token/src/entrypoint.rs

```diff
@@ -14,218 +13,245 @@ default_panic_handler!();
+    match *discriminator {
         // 0 - InitializeMint
-        TokenInstruction::InitializeMint => {
+        0 => {
             #[cfg(feature = "logging")]
             pinocchio::msg!("Instruction: InitializeMint");
 
             process_initialize_mint(accounts, instruction_data)
         }
+        // 1 - InitializeAccount
+        1 => {
+            #[cfg(feature = "logging")]
+            pinocchio::msg!("Instruction: InitializeAccount");
 
+            process_initialize_account(accounts)
+        }
         // 3 - Transfer
-        TokenInstruction::Transfer => {
+        3 => {
             #[cfg(feature = "logging")]
             pinocchio::msg!("Instruction: Transfer");
 
             process_transfer(accounts, instruction_data)
         }
         // 7 - MintTo
-        TokenInstruction::MintTo => {
+        7 => {
             #[cfg(feature = "logging")]
             pinocchio::msg!("Instruction: MintTo");
 
             process_mint_to(accounts, instruction_data)
         }
         // 9 - CloseAccount
-        TokenInstruction::CloseAccount => {
+        9 => {
             #[cfg(feature = "logging")]
             pinocchio::msg!("Instruction: CloseAccount");
 
             process_close_account(accounts)
         }
+        // 16 - InitializeAccount2
+        16 => {
+            #[cfg(feature = "logging")]
+            pinocchio::msg!("Instruction: InitializeAccount2");
+
+            process_initialize_account2(accounts, instruction_data)
+        }
         // 18 - InitializeAccount3
-        TokenInstruction::InitializeAccount3 => {
+        18 => {
             #[cfg(feature = "logging")]
             pinocchio::msg!("Instruction: InitializeAccount3");
 
             process_initialize_account3(accounts, instruction_data)
         }
         // 20 - InitializeMint2
-        TokenInstruction::InitializeMint2 => {
+        20 => {
             #[cfg(feature = "logging")]
             pinocchio::msg!("Instruction: InitializeMint2");
 
             process_initialize_mint2(accounts, instruction_data)
         }
-        _ => process_remaining_instruction(accounts, instruction_data, instruction),
+        d => inner_process_remaining_instruction(accounts, instruction_data, d),
     }
 }
 
-/// Process the remaining instructions.
+/// Process a remaining "regular" instruction.
 ///
-/// This function is called by the `process_instruction` function if the discriminator
+/// This function is called by the [`inner_process_instruction`] function if the discriminator
 /// does not match any of the common instructions. This function is used to reduce the
-/// overhead of having a large `match` statement in the `process_instruction` function.
-fn process_remaining_instruction(
+/// overhead of having a large `match` statement in the [`inner_process_instruction`] function.
+fn inner_process_remaining_instruction(
     accounts: &[AccountInfo],
     instruction_data: &[u8],
-    instruction: TokenInstruction,
```

Comment:
Same here: was this supposed to be reverted?

Reply:
Yes, it is more efficient to use the `u8` value directly.